src/EventSubscriber/XLocationIdResponseSubscriber.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformHttpCacheBundle\EventSubscriber;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Rewrites the X-Location-Id HTTP header.
+ *
+ * This is a BC layer for custom controllers (including REST server) still
+ * using X-Location-Id header which is now deprecated. For
+ * full value of tagging, see docs/using_tags.md for how to take advantage of the
+ * system.
+ */
+class XLocationIdResponseSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $tagHeader;
+
+    /**
+     * @var LoggerInterface|null
+     */
+    private $logger;
+
+    public function __construct($tagHeader, LoggerInterface $logger = null)
+    {
+        $this->tagHeader = $tagHeader;
+        $this->logger = $logger;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [KernelEvents::RESPONSE => ['rewriteCacheHeader', -5]];
+    }
+
+    public function rewriteCacheHeader(FilterResponseEvent $event)
+    {
+        $response = $event->getResponse();
+        if (!$response->headers->has('X-Location-Id')) {
+            return;
+        }
+
+        if ($this->logger instanceof LoggerInterface) {
+            $this->logger->notice(
+                'X-Location-Id rewritten to use ezplatform-http-cache tag header, see \<bundle\>/docs/using_tags.md for more info'
+            );
+        }
+
+        // Map the tag, and even if it was not officially supported, make sure to handle comma separated values as was possible with Varnish
+        $tags = array_map(
+            function ($id) {return 'location-' . trim($id);},
+            explode(',', $response->headers->get('X-Location-Id'))
+        );
+
+        // Check if there is already exiting tags and merge them
+        if ($response->headers->has($this->tagHeader)) {
+            $tags += $response->headers->get($this->tagHeader, null, false);
+        }
+
+        // Set headers
+        // @todo we need to use abstract tag writer to also be able to support Fastly
```

Comment:
What does the abstract tag writer do that we don't do here ? Since we use the header name as set in the fos configuration, it should work with Fastly, should it not ?

Reply:
I meant to check with @vidarl, but the value is different in Fastly examples, it's separated by space: https://docs.fastly.com/guides/purging/getting-started-with-surrogate-keys#understanding-surrogate-keys

So if Fastly does not support what we currently do*, then we need to come up with something for abstracting this. And we might actually have to do that anyway, as if we want to promote usage of FosHttpCache TagsHandler for instance for custom tag usage, without hard-coding xkey header, it currently [always writes comma separated values](https://github.com/FriendsOfSymfony/FOSHttpCache/blob/1.4/src/Handler/TagHandler.php#L91-L94) in 1.x _(2.x to but it has [a formater object](https://github.com/FriendsOfSymfony/FOSHttpCache/search?utf8=%E2%9C%93&q=getTagsHeaderValue&type=) you can change to something else)_.


* Respond with several headers to comply with xkey doc: https://github.com/varnish/varnish-modules/blob/master/docs/vmod_xkey.rst